driver/src/models/mod.rs

```diff
@@ -20,6 +20,10 @@ pub trait RollingHashable {
     fn rolling_hash(&self, nonce: i32) -> H256;
 }
 
+pub trait ConcatenatingHashable {
+    fn concating_hash(&self, init_hash: H256) -> H256;
```

Comment:
here too; although this is somewhat different. My best guess here as a suggestion would be `concatenate_hash` or simply `concat_hash`.